src/main/java/at/hannibal2/skyhanni/features/misc/customscoreboard/CustomScoreboard.kt

```diff
@@ -0,0 +1,127 @@
+
+import at.hannibal2.skyhanni.SkyHanniMod
+import at.hannibal2.skyhanni.events.GuiPositionMovedEvent
+import at.hannibal2.skyhanni.events.GuiRenderEvent
+import at.hannibal2.skyhanni.events.LorenzTickEvent
+import at.hannibal2.skyhanni.utils.ChatUtils
+import at.hannibal2.skyhanni.utils.LorenzUtils
+import at.hannibal2.skyhanni.utils.RenderUtils.HorizontalAlignment
+import at.hannibal2.skyhanni.utils.RenderUtils.renderStringsAlignedWidth
+import at.hannibal2.skyhanni.utils.TabListData
+import net.minecraftforge.client.GuiIngameForge
+import net.minecraftforge.client.event.RenderGameOverlayEvent
+import net.minecraftforge.fml.common.eventhandler.SubscribeEvent
+
+typealias ScoreboardElementType = Pair<String, HorizontalAlignment>
+
+class CustomScoreboard {
+    private val config get() = SkyHanniMod.feature.gui.customScoreboard
+    private var display = emptyList<ScoreboardElementType>()
+    private var cache = emptyList<ScoreboardElementType>()
+    private val guiName = "Custom Scoreboard"
+
+    @SubscribeEvent
+    fun onRenderOverlay(event: GuiRenderEvent.GuiOverlayRenderEvent) {
+        if (!isEnabled()) return
+        if (display.isEmpty()) return
+
+        RenderBackground().renderBackground()
+
+        if (!TabListData.fullyLoaded && config.displayConfig.cacheScoreboardOnIslandSwitch && cache.isNotEmpty()) {
+            config.position.renderStringsAlignedWidth(cache, posLabel = guiName)
+        } else {
+            config.position.renderStringsAlignedWidth(display, posLabel = guiName)
+
+            cache = display
+        }
+    }
+
+    @SubscribeEvent
+    fun onGuiPositionMoved(event: GuiPositionMovedEvent) {
+        if (event.guiName == guiName) {
+            val alignmentConfig = config.displayConfig.alignment
+            if (alignmentConfig.alignRight || alignmentConfig.alignCenterVertically) {
+                alignmentConfig.alignRight = false
+                alignmentConfig.alignCenterVertically = false
+                ChatUtils.chat("Disabled Custom Scoreboard auto-alignment.")
+            }
+        }
+    }
+
+    @SubscribeEvent
+    fun onTick(event: LorenzTickEvent) {
+        if (!isEnabled()) return
+
+        // Creating the lines
+        if (event.isMod(5)) {
+            display = createLines()
+        }
+
+        // Remove Known Lines, so we can get the unknown ones
+        UnknownLinesHandler.handleUnknownLines()
+    }
+
+    private fun createLines() = buildList<ScoreboardElementType> {
+        val lineMap = ScoreboardElement.entries.associate {
+            it.ordinal to
+                if (it.isVisible()) it.getPair() else listOf("<hidden>" to HorizontalAlignment.LEFT)
+        }
+
+        return formatLines(lineMap)
+    }
+
+    private fun formatLines(lineMap: Map<Int, List<ScoreboardElementType>>): List<ScoreboardElementType> {
+        return buildList {
+            for (element in config.scoreboardEntries) {
+                val line = lineMap[element.ordinal] ?: continue
```

Comment:
idk why you made this so complicated.
In scordeboardElements:
```
    fun getVisiblePair() = if (isVisible()) getPair() else listOf("<hidden>" to HorizontalAlignment.LEFT)
```
```suggestion
    private fun createLines() = buildList<ScoreboardElementType> {
        for (element in config.scoreboardEntries) {
            val line = element.getVisiblePair()
```